nvim/session.lua

```diff
@@ -1,6 +1,7 @@
 require('coxpcall')
 local uv = require('luv')
 local MsgpackRpcStream = require('nvim.msgpack_rpc_stream')
+local unpack = unpack or table.unpack -- luacheck: globals unpack (compatibility with Lua 5.1)
```

Comment:
If we're going to start doing this in lua-compat then we might as well stub out a `compat.lua` similar to Nvim: https://github.com/neovim/neovim/blob/4946751906370134cd02e8da0736bef221171172/runtime/lua/vim/compat.lua#L1


Also I'll mention again that if we're going to spend time on lua-client then maybe it makes sense to migrate it into the `neovim` repo instead of having a separate repo.

Reply:
> stub out a `compat.lua`

Maybe, yes - but might be overkill (since the patch/fix is rather small as of now), but I can add a module for it.

> Also I'll mention again that if we're going to spend time on lua-client then maybe it makes sense to migrate it into the `neovim` repo instead of having a separate repo.

Yes.  But I want to not create too many moving pieces here in this regard.  I think it makes sense to fix it here ("upstream") first.

Reply:
> Maybe, yes - but might be overkill (since the patch/fix is rather small as of now), but I can add a module for it.

It always starts out small. The problem is tracking down all the little pieces later when it would be "big".

Reply:
> Also I'll mention again that if we're going to spend time on lua-client then maybe it makes sense to migrate it into the `neovim` repo instead of having a separate repo.

Is there a plan for this (e.g. an issue)?  It is not clear to me how this should be done / approached.  I think it would be good to have an issue about it in neovim probably?

Reply:
There's no tracking issue, would be good to have one.

Reply:
@justinmk ok, please create one then.  I might come back to it otherwise myself, but I assume you have a better plan for it already.

Reply:
https://github.com/neovim/neovim/issues/11474